### PR TITLE
feat(sheets-ui): support canvas float dom

### PR DESCRIPTION
**Pull request by weird94**
close #xxx

## Pull Request Checklist

- [x] Related tickets or issues have been linked in the PR description (or missing issue).
- [x] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [x] Unit tests have been added for the changes (if applicable).
- [x] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Comment by weird94** [comment]
https://github.com/dream-num/univer/pull/2343#issuecomment-2137208048

File: packages/sheets-drawing-ui/src/plugin.ts [hunk 1]
```diff
@@ -50,7 +51,7 @@ export class UniverSheetsDrawingUIPlugin extends Plugin {
         const dependencies: Dependency[] = [
 
             // services
-
+            [SheetCanvasFloatDomManagerService],
```
Comment (wzhudev) [inline packages/sheets-drawing-ui/src/plugin.ts:54]: 对于每一种 floating 元素都有一个对应的 manager？

File: packages/ui/src/services/dom/canvas-dom-layer.service.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    get domLayers() {
+        return Array.from(this._domLayerMap.entries());
+    }
+
+    private _notice() {
+        this._domLayers$.next(Array.from(this._domLayerMap.entries()));
+    }
+
+    addFloatDom(item: IFloatDom) {
```
Comment (wzhudev) [inline packages/ui/src/services/dom/canvas-dom-layer.service.ts:54]: Return an `IDisposable`. `RemoveFloatom` is not necessary.

File: packages/ui/src/views/App.tsx [hunk 3]
```diff
@@ -151,6 +152,7 @@ export function App(props: IUniverAppProps) {
                                 onContextMenu={(e) => e.preventDefault()}
                             >
                                 <CanvasPopup />
+                                <FloatDom />
```
Comment (wzhudev) [inline packages/ui/src/views/App.tsx:155]: 应该用注册的方式渲染出来，避免在 app.tsx 里面写死，包括 canvas popup 也是一样

**Review by wzhudev — CHANGES_REQUESTED** [review]

File: packages/ui/src/services/dom/canvas-dom-layer.service.ts (hunk 2 above)
Comment (weird94) [reply packages/ui/src/services/dom/canvas-dom-layer.service.ts:54]: fixed

**Review by wzhudev — APPROVED** [review]
LGTM

**Pull request merged**